src/api/services/auditTrailService.ts

```diff
@@ -1,279 +1,21 @@
-    };
-
-    return await AuditTrail.query().insert(sharingAuditTrail);
-  } catch (error) {
-    const { errorLogger } = getLoggers();
-    errorLogger.error(`Audit trails inserted failed: ${error}`, traceId);
-    throw error;
-  }
-};
-
-export const insertKeyPairAuditTrails = async (
-  participant: Participant,
-  userId: number,
-  userEmail: string,
-  action: AuditAction,
-  name: string,
-  disabled: boolean,
-  traceId: string
-) => {
-  try {
-    const keyPairAuditTrail: Omit<AuditTrailDTO, 'id'> = {
-      userId,
-      userEmail,
-      event: AuditTrailEvents.ManageKeyPair,
-      eventData: {
-        siteId: participant.siteId!,
-        action,
-        name,
-        disabled,
-        participantId: participant.id,
-      },
-      succeeded: false,
-    };
-
-    return await AuditTrail.query().insert(keyPairAuditTrail);
-  } catch (error) {
-    const { errorLogger } = getLoggers();
-    errorLogger.error(`Audit trails inserted failed: ${error}`, traceId);
-    throw error;
-  }
-};
-
-export const insertDomainNamesAuditTrails = async (
-  participant: Participant,
-  userId: number,
-  userEmail: string,
-  action: AuditAction,
-  domainNames: string[],
-  traceId: string
+  operation: () => Promise<T>
 ) => {
+  const insertedAuditTrail = await AuditTrail.query().insert({ ...auditTrail, succeeded: false });
   try {
-    const domainNamesAuditTrail: Omit<AuditTrailDTO, 'id'> = {
-      userId,
-      userEmail,
-      event: AuditTrailEvents.UpdateDomainNames,
-      eventData: {
-        siteId: participant.siteId!,
-        action,
-        participantId: participant.id,
-        domainNames,
-      },
-      succeeded: false,
-    };
-
-    return await AuditTrail.query().insert(domainNamesAuditTrail);
+    const result = await operation();
+    await AuditTrail.query().patch({ succeeded: true }).where('id', insertedAuditTrail.id);
+    return result;
   } catch (error) {
     const { errorLogger } = getLoggers();
-    errorLogger.error(`Audit trails inserted failed: ${error}`, traceId);
+    errorLogger.error(`${auditTrail.event} failed with error: ${error}`, traceId);
     throw error;
   }
 };
-
-export const insertAppNamesAuditTrails = async (
-  participant: Participant,
-  userId: number,
-  userEmail: string,
-  action: AuditAction,
-  appNames: string[],
-  traceId: string
-) => {
-  try {
-    const appNamesAuditTrail: Omit<AuditTrailDTO, 'id'> = {
-      userId,
-      userEmail,
-      event: AuditTrailEvents.UpdateAppNames,
-      eventData: {
-        siteId: participant.siteId!,
-        action,
-        participantId: participant.id,
-        appNames,
-      },
-      succeeded: false,
-    };
-
-    return await AuditTrail.query().insert(appNamesAuditTrail);
-  } catch (error) {
-    const { errorLogger } = getLoggers();
-    errorLogger.error(`Audit trails inserted failed: ${error}`, traceId);
-    throw error;
-  }
-};
-
-export const insertApproveAccountAuditTrail = async (
```

Comment:
This function and the below function had a slightly different pattern, where they accepted an object rather than a list of parameters. This was more elegant than all the above functions, hence the new pattern does not look as impressive where `insertApproveAccountAuditTrail` and `insertAddParticipantAuditTrail` were previously used.